src/jit/codegencommon.cpp

```diff
@@ -10317,10 +10333,23 @@ void CodeGen::genFuncletProlog(BasicBlock* block)
 
     compiler->unwindBegProlog();
 
+    inst_RV(INS_push, REG_FPBASE, TYP_REF);
+    compiler->unwindPush(REG_FPBASE);
+
     // TODO Save callee-saved registers
 
     // This is the end of the OS-reported prolog for purposes of unwinding
     compiler->unwindEndProlog();
+
+    // Restore frame pointer
+    // TODO We will need changes here when we introduce PSPSym
+    inst_RV_RV(INS_mov, REG_FPBASE, REG_ARG_0);
```

Comment:
I simply tested by manually setting ESP larger by 4 on GDB and it works! and I made a patch for that. https://github.com/wateret/coreclr/commit/b5c4f4502a9eae066103578175d2d4ab622da01e is that what you meant? It passes the test case I gave.

BTW I was looking at `ThrowExceptionFromContextInternal`, is there some meaning for pushing Context_Eip before calling into `ThrowExceptionHelper`? I see that some kind of return address should be pushed since calling `ThrowExceptionHelper` with `jmp` instruction. However it looks like doing nothing since it will never return.